src/components/media-content/edit-media-content-view.tsx

```diff
@@ -54,7 +55,7 @@ const EditMediaContentView = ({
       let thumbnailUrl = thumbnail ? await uploadFile(thumbnail) : undefined;
 
       await editMediaContent(
-        { ...mediaContent, ...result, topicId: topicId as string, unitId: unitId as string },
+        { ...mediaContent, ...result, topicId: topicId || mediaContent.topicId, unitId: unitId || mediaContent.unitId },
```

Comment:
Yeah, they're also marked as optional on the backend, but it seems `zfd.optional()` expects the field to either be `undefined` or just absent from the object, but not `null`. (https://gist.github.com/ciiqr/ee19e9ff3bb603f8c42b00f5ad8c551e)

In this case with the changes it will resolve to either the ID string or undefined when absent